core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -117,9 +126,23 @@
   }
 
   private final class ContextCancellationListener implements CancellationListener {
+    private Listener<RespT> observer;
```

Comment:
Annotate it as nullable, and add a comment saying that this observer is non-null only if we want to delay the cancellation of stream in case of deadline-exceeded.